src/main/java/no/unit/nva/doi/transformer/language/LanguageMapper.java

```diff
@@ -0,0 +1,144 @@
+package no.unit.nva.doi.transformer.language;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+import no.unit.nva.doi.transformer.language.exceptions.LanguageUriNotFoundException;
+
+public final class LanguageMapper {
+
+    public static final String FIELD_DELIMITER = "\t";
+    private static final Map<String, URI> ISO2URI = isoToUri(Path.of("languages", "lexvo-iso639-3.tsv"));
+    private static final Map<URI, String> URI2ISO = uriToIso(Path.of("languages", "lexvo-iso639-3.tsv"));
+    public static final String ERROR_READING_FILE = "Could not read resource file:";
+    public static final String URI_NOT_FOUND_ERROR = "Could not find a URI for the language:";
+
+    private LanguageMapper() {
+    }
+
+    /**
+     * Map an ISO639-3 language identifier to a Language URI.
+     *
+     * @param iso6393 An ISO639-3 identifier.
+     * @return a language URI if this mapping is available or an empty {@link Optional} if there is no such mapping.
+     */
+    public static Optional<URI> getUriOpt(String iso6393) {
+        if (iso6393 != null) {
+            return Optional.ofNullable(ISO2URI.get(iso6393));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Returns an ISO639-3 code for the specified URI.
+     *
+     * @param uri A language URI
+     * @return An ISO639-3 code or empty optional if there is no mapping for the input URI
+     */
+    public static Optional<String> getIsoOpt(URI uri) {
+        if (uri != null) {
+            return Optional.ofNullable(URI2ISO.get(uri));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Map an ISO639-3 language identifier to a Language URI.
+     *
+     * @param iso6393 An ISO639-3 identifier.
+     * @return a language URI if this mapping is available or an empty {@link Optional} if there is no such mapping.
+     * @throws LanguageUriNotFoundException when there is no mapping between the input string the available mappings.
+     */
+    public static URI getUri(String iso6393) throws LanguageUriNotFoundException {
```

Comment:
Maybe add what we are getting URI from in method name?

Reply:
done